### PR TITLE
Collect snapshot chunks from peers

### PR DESCRIPTION
**Pull request by Remagpie**

### CODE REVIEW

File: sync/src/block/extension.rs [hunk 1]
```diff
@@ -193,6 +234,14 @@ impl Extension {
                 })
                 .collect();
 
+            let chunk_requests: Vec<RequestMessage> = requests
+                .iter()
+                .filter_map(|r| match r {
+                    (_, RequestMessage::StateChunk(..)) => Some(r.1.clone()),
```
Comment (foriequal0) [inline sync/src/block/extension.rs]: Binding is more clear.
`(_, msg@RequestMessage::StateChunk(..)) => Some(msg.clone()),`

File: util/merkle/src/snapshot/mod.rs [hunk 2]
```diff
@@ -91,6 +90,12 @@ impl<'a> Restore<'a> {
     }
 }
 
+impl std::fmt::Debug for Restore {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        write!(f, "Restore {{ pending: {:?} }}", self.pending)
```
Comment (foriequal0) [inline util/merkle/src/snapshot/mod.rs]: You can use this: https://doc.rust-lang.org/std/fmt/struct.DebugStruct.html
```rust
        fmt.debug_struct("Restore")
           .field("pending", &self.pending)
           .field("unresolved", "<...>")
           .finish()
```

File: sync/src/block/extension.rs (hunk 1 above)
Comment (Remagpie) [reply sync/src/block/extension.rs]: Fixed.

File: util/merkle/src/snapshot/mod.rs (hunk 2 above)
Comment (Remagpie) [reply util/merkle/src/snapshot/mod.rs]: Fixed.

**Pull request merged**